include/aspect/boundary_traction/interface.h

```diff
@@ -125,7 +125,181 @@ namespace aspect
          */
         const GeometryModel::Interface<dim> *geometry_model;
     };
+    template <int dim>
+    class Manager : public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Destructor. Made virtual since this class has virtual member
+         * functions.
+         */
+        ~Manager () override;
 
+        /**
+         * A function that is called at the beginning of each time step and
+         * calls the corresponding functions of all created plugins.
+         *
+         * The point of this function is to allow complex boundary traction
+         * models to do an initialization step once at the beginning of each
+         * time step. An example would be a model that needs to call an
+         * external program to compute the traction change at a boundary.
+         */
+        virtual
+        void
+        update ();
+
+        /**
+         * A function that calls the boundary_traction functions of all the
+         * individual boundary traction objects and uses the stored operators
+         * to combine them.
+         */
+        Tensor<1,dim>
+        boundary_traction (const types::boundary_id boundary_indicator,
```

Comment:
Here you will have to change something compared to the velocity manager. Take a look at line 100 of this file, the `boundary_traction` function has a slightly different interface. In particular it takes a third argument that indicates the normal direction on the current boundary. You will have to add this here as an additional argument as well, otherwise the manager cannot call the respective function of the individual plugins.